support/jwt/src/main/kotlin/com/xorker/xpr/support/jwt/JwtProvider.kt

```diff
@@ -0,0 +1,95 @@
+package com.xorker.xpr.support.jwt
+
+import com.fasterxml.jackson.core.JsonParseException
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.DatabindException
+import com.fasterxml.jackson.databind.ObjectMapper
+import io.jsonwebtoken.Claims
+import io.jsonwebtoken.JwtException
+import io.jsonwebtoken.JwtParser
+import io.jsonwebtoken.Jwts
+import java.security.PublicKey
+import java.time.LocalDateTime
+import java.time.LocalDateTime.now
+import java.time.ZoneId
+import java.util.*
+import javax.crypto.SecretKey
+
+object JwtProvider {
+    private val objectMapper = ObjectMapper()
+    private val typeReference = object : TypeReference<Map<String, String>>() {}
+
+    fun generate(id: String, expiration: LocalDateTime, key: SignatureKey): String {
+        return Jwts.builder()
+            .issuedAt(now().toDate())
+            .expiration(expiration.toDate())
+            .subject(id)
+            .signWith(key.getKey())
+            .compact()
+    }
+
+    fun validate(token: String, key: SignatureKey): Boolean {
+        try {
+            val parser = when (val signatureKey = key.getKey()) {
+                is SecretKey -> generateParser(signatureKey)
+                is PublicKey -> generateParser(signatureKey)
+                else -> return false
+            }
+            parser.parseSignedClaims(token)
+            return true
+        } catch (e: JwtException) {
+            return false
+        }
+    }
+
+    fun validateClaimsWith(iss: String, aud: String, token: String, key: SignatureKey): Boolean {
+        val claims = getClaims(token, key) ?: return false
+        return !(claims.issuer != iss || !claims.audience.contains(aud))
```

Comment:
p4. not 이 너무 많아서 오히려 가독성이 떨어지는 것 같아요
```suggestion
        return claims.issuer == iss && claims.audience.contains(aud)
```